src/controller/CHIPDeviceController.h

```diff
@@ -167,6 +166,8 @@ class DLL_EXPORT DeviceController : public AbstractDnssdDiscoveryController
 
     CHIP_ERROR GetPeerAddressAndPort(PeerId peerId, Inet::IPAddress & addr, uint16_t & port);
 
+    CHIP_ERROR GetPeerAddressInterfaceAndPort(PeerId peerId, Inet::IPAddress & addr, Inet::InterfaceId & iface, uint16_t & port);
```

Comment:
Please do not add APIs without documentation.  The documentation should explain when it's OK to call this API and expect a useful answer, at the very least.  Because this API needs to be used in a _very_ particular way that is totally not obvious from here.